clients/baseclient/client_util_test.go

```diff
@@ -0,0 +1,108 @@
+package baseclient
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ClientUtil", func() {
+	Describe("ShouldRetry", func() {
+		Context("when zero value pointer error(uninitialized value - nil) is passed as argument", func() {
+			It("Retry operation call shouldn't be made", func() {
+				result := shouldRetry(nil)
+				Expect(result).To(Equal(false))
+			})
+		})
+
+		Context("when the backend returns status code with first digit 1 (1xx)", func() {
+			It("Retry operation call is expected to be made", func() {
+				err := ClientError{102, "Processing", "The request has been accepted but it's not yet complete"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(true))
+			})
+		})
+
+		Context("when the backend returns status code with first digit 2 (2xx)", func() {
+			It("Retry operation call shouldn't be made", func() {
+				err := ClientError{200, "OK", "The request has succeeded"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(false))
+			})
+		})
+
+		Context("when the backend returns status code with first digit 3 (3xx)", func() {
+			It("Retry operation call is expected to be made", func() {
+				err := ClientError{301, "Moved Permanently", "URI of requested resource has been changed"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(true))
+			})
+		})
+
+		Context("when the backend returns status code with first digit 4 (4xx)", func() {
+			It("Retry operation call is expected to be made", func() {
+				err := ClientError{404, "Not Found", "The server cannot find requesred resource"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(true))
+			})
+		})
+
+		Context("when the backend returns status code with first digit 5 (5xx)", func() {
+			It("Retry operation call is expected to be made", func() {
+				err := ClientError{500, "Internal Server Error", "The server got an invalid response"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(true))
+			})
+		})
+
+		Context("when the passed error is not of type 'ClientError'", func() {
+			It("Retry operation call shouldn't be made", func() {
+				err := MockError{999, "Not ClientError"}
+				result := shouldRetry(&err)
+				Expect(result).To(Equal(false))
+			})
+		})
+	})
+})
+
+var _ = Describe("ClientUtil", func() {
+	Describe("CallWithRetry", func() {
+		Context("when retry operation call is Not needed", func() {
+			It("Just retrun required response", func() {
+				getInterface := func() (interface{}, error) {
```

Comment:
rename 'getInterface' with something a bit more descriptive. e.g. 'mockCallback' or 'basicCallback' or 'failingCallback' for example below

Reply:
Absolutely agree with you!